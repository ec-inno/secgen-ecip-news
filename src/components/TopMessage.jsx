import React from 'react';

import getCurrentLanguage from '../utils/getCurrentLanguage';
import getDefaultLanguage from '../utils/getDefaultLanguage';

const TopMessage = ({ location }) => {
  const language = getCurrentLanguage(location) || getDefaultLanguage();

  const data = require(`../data/topmessage/${language}.json`);
  const { message } = data;

  return (
    <div className="eci-menu" id="top">
      <div className="ecl-container">{message}</div>
    </div>
  );
};

export default TopMessage;
