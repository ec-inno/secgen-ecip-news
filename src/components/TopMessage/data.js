const data = {
  bg: 'An official website of the European union (BG)',
  en: 'An official website of the European union',
  fr: 'An official website of the European union (FR)',
};

export default data;
