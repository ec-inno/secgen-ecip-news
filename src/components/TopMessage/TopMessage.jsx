import React from 'react';

import data from './data';

const TopMessage = ({ currentLanguage }) => {
  const message = data[currentLanguage];

  return (
    <div className="eci-menu" id="top">
      <div className="ecl-container">{message}</div>
    </div>
  );
};

export default TopMessage;
