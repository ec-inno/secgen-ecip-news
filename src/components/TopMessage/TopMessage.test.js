import React from 'react';
import renderer from 'react-test-renderer';
import TopMessage from './TopMessage';

describe('TopMessage', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TopMessage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
