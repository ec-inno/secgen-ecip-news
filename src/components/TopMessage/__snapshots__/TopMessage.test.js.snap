// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMessage renders correctly 1`] = `
<div
  className="eci-menu"
  id="top"
>
  <div
    className="ecl-container"
  >
    An official website of the European union
  </div>
</div>
`;
